### DISCUSSION TITLE
What`s the difference between deault and insert_default when define column with mapped_column

### DISCUSSION THREAD

**Discussion by TMGT** [sqlalchemy/sqlalchemy#11165] [Usage Questions] [answered]
Hi there, I am a beginer of SQLAlchemy, I am confused about the difference between deault  and insert_default when define column with mapped_column, I know that both of them will set a value to a column when create a new row and the column don`t have a given value , but what is difference between them.I already read the document of default and insert_default, but I still cannot understand that. I will appreciate that if anyone can help me make that clear.

**Comment by CaselIT** [comment, marked as answer]
Hi,

They are different only when using mapped as dataclass to make the orm classes also dataclass.
In that case `default` behaves like the detaclass default would, meaning it's the value taken by the attribute when you instantiate without a value; `insert_default` instead is the default value that will be used when inserting a row to the db, and this can be a literal text sql or a sql function call etc.
Usually they are mutually exclusive

When not using a dataclass both behave like insert_default, meaning that it's used only when inserting a row to the db

**Reply by TMGT** [reply]
Oh, thank you very much! You pointed out the point, now I see it, thank you again for your kindly help.

**Reply by CaselIT** [reply]
Do you have a suggestion to improve the docs? Maybe we could do a better job indicating that the two are only relevant when dataclasses are used?

**Reply by TMGT** [reply]
I donot have too much suggestion about that, maybe some example code would be helpful? Recently, I read the doc again, i think user who are working with dataclass are able to understand that easily, but user like me who are not familiar with dataclass or combine sqlalchemy and dataclass may have some diffcult to understand that, so i think maybe some example code would be helpful, although it is clear enough for the exprienced user.

**Reply by CaselIT** [reply]
Ok, thanks for the feedback!
